src/router/index.ts

```diff
@@ -62,31 +62,41 @@ export function projectLinkHref(
 
 export function updateProjectRoute(
   projectRouteParams: Partial<ProjectsParams>,
+  e?: MouseEvent,
 ) {
   const activeRoute = get(activeRouteStore);
 
   if (activeRoute.resource === "projects") {
-    const updatedRoute = createProjectRoute(activeRoute, projectRouteParams);
-    push(updatedRoute);
+    if (e?.button === 1) {
+      const href = projectLinkHref(projectRouteParams);
+      window.open(href, "_blank");
+    } else {
+      const updatedRoute = createProjectRoute(activeRoute, projectRouteParams);
+      push(updatedRoute);
+    }
   } else {
     throw new Error(
       "Don't use project specific navigation outside of project views",
     );
   }
 }
 
-export const push = (newRoute: Route): void => {
+export const push = (newRoute: Route, e?: MouseEvent): void => {
   const history = get(historyStore);
 
-  // Limit history to a maximum of 10 steps. We shouldn't be doing more than
-  // one subsequent pop() anyway.
-  historyStore.set([...history, newRoute].slice(-10));
+  if (e?.button === 1) {
```

Comment:
Hey @rudolfs this PR is working great for adding middle click behaviour and I'm happy to be able to remove a lot of event dispatches.

The thing I'm not 100% certain is if the checking here against `e.button` is the best one.. since users are able to change their pointer devices configuration so 1 could be left click or right click or whatever else click..
The other way would be to start with `onauxclick` and later see if some better solution comes up for Safari users, wdyt?

Reply:
I guess we can try this and see if people complain? 🤷
Don't know what the best solution is here.

Reply:
Ok cool, I can live with that